ddtrace/debugging/_expressions.py

```diff
@@ -196,7 +196,14 @@ def _compile_direct_operation(ast):
 
 
 def _call_function(func, *args):
-    return [Instr("LOAD_CONST", func)] + list(chain(*args)) + [Instr("CALL_FUNCTION", len(args))]
+    # type: (Callable, List[Instr]) -> List[Instr]
+    if PY < (3, 11):
+        return [Instr("LOAD_CONST", func)] + list(chain(*args)) + [Instr("CALL_FUNCTION", len(args))]
+    return (
+        [Instr("PUSH_NULL"), Instr("LOAD_CONST", func)]
+        + list(chain(*args))
+        + [Instr("PRECALL", len(args)), Instr("CALL", len(args))]
+    )
```

Comment:
Deduplicating these two conditional branches could make this code easier to understand.
```suggestion
    # type: (Callable, List[Instr]) -> List[Instr]
    instructions = [Instr("LOAD_CONST", func)] + list(chain(*args)) + [Instr("CALL_FUNCTION", len(args))]
    if PY < (3, 11):
        instructions.insert(0, Instr("PUSH_NULL"))
        instructions.pop()
        instructions.extend([Instr("PRECALL", len(args)), Instr("CALL", len(args))])
    return instructions
```
It's debatable whether this is easier to read. For me personally it would have driven the point home that the core logic between these two branches is the same.

Reply:
I was in two minds about the approach to take here, but went with the repetition to give each branch the full set of instructions required to implement the operation and I thought it was clearer this way